### ISSUE TITLE
False positives for @apply mixins with root-no-standard-properties

### ISSUE THREAD

**Issue by kevinSuttle** [stylelint/stylelint#1789]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Allowing mixin definitions (for later use with `@apply`) inside of `:root` triggers a warning. 

> Which rule, if any, is this issue related to?

`root-no-standard-properties`

> What CSS is needed to reproduce this issue?

```css
:root {
  --ThemeAnimationStyles {
      transition-duration: .2s;
      transition-timing-function: cubic-bezier(.075, .820, .165, 1);
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "root-no-standard-properties": true
  }
}
```

> Which version of stylelint are you using?

`7.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Visual Studio Code plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Maybe? 

> What did you expect to happen?

 "No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

"The following warnings were flagged:"

```
Unexpected standard property:  transition-duration
```

**Comment by jeddy3** [comment]
@kevinSuttle Thank you for the detailed report.

I think the issue is [here](https://github.com/stylelint/stylelint/blob/master/src/rules/root-no-standard-properties/index.js#L28) i.e. we use `walkDecls` which is a deep iterator. Swapping this to `each` should resolve the issue. Would you like to contribute a fix?

**Comment by kevinSuttle** [comment]
Swapping `rule.walkDecls(function (decl)` to `rule.each(function (decl) {` ?

**Comment by jeddy3** [comment]
Yes, I think [that'll do it](http://api.postcss.org/AtRule.html#each).

**Comment by kevinSuttle** [comment]
Opened PR: https://github.com/stylelint/stylelint/pull/1794

**Comment by davidtheclark** [comment]
@kevinSuttle What happens if you use a colon after the property in the declaration, as it seems you're supposed to from the spec & plugin README?
